src/components/MenuItem.js

```diff
@@ -87,14 +87,20 @@ const MenuItem = (props) => {
     );
     const descriptionTextStyle = StyleUtils.combineStyles([
         styles.textLabelSupporting,
-        props.icon ? styles.ml3 : undefined,
+        props.icon && !_.isArray(props.icon) ? styles.ml3 : undefined,
         styles.lineHeightNormal,
         props.title ? descriptionVerticalMargin : undefined,
         props.descriptionTextStyle,
         isDeleted ? styles.offlineFeedback.deleted : undefined,
     ]);
 
     const fallbackAvatarSize = props.viewMode === CONST.OPTION_MODE.COMPACT ? CONST.AVATAR_SIZE.SMALL : CONST.AVATAR_SIZE.DEFAULT;
+    const stylesWithoutPadding = _.isArray(props.style)
+        ? _.map(props.style, (style) => _.omit(style, ['padding', 'paddingVertical', 'paddingHorizontal']))
+        : _.omit(props.style, ['padding', 'paddingVertical', 'paddingHorizontal']);
+    const paddingStyles = _.isArray(props.style)
+        ? _.map(props.style, (style) => _.pick(style, ['padding', 'paddingVertical', 'paddingHorizontal']))
+        : _.pick(props.style, ['padding', 'paddingVertical', 'paddingHorizontal']);
```

Comment:
Can you please get rid of this? I don't see the need for a new View (or new padding styles)

Reply:
In this case we should just change the default style to flex column (add it below props.style) and in the new view use flex row.